rpc/grpc/api.go

```diff
@@ -37,3 +49,345 @@ func (bapi *broadcastAPI) BroadcastTx(ctx context.Context, req *RequestBroadcast
 		},
 	}, nil
 }
+
+type BlockAPI struct {
+	sync.Mutex
+	heightListeners      map[chan NewHeightEvent]struct{}
+	newBlockSubscription eventstypes.Subscription
+}
+
+func NewBlockAPI() *BlockAPI {
+	return &BlockAPI{
+		heightListeners: make(map[chan NewHeightEvent]struct{}, 1000),
+	}
+}
+
+func (blockAPI *BlockAPI) StartNewBlockEventListener(ctx context.Context) {
```

Comment:
the issue with this is that the cosmos-sdk doesn't listen for errors after the server starts listening (for about 5sec or so). Making it listen would require more changes, which I tried to reduce given we want to reduce diff with upstream. 

Current way it will log errors but not stop the node or anything.

Reply:
having this error in the method is cleaner than adding logs for every return. There are also a few `return`s that don't error. 

It also looks like this is called in cometbft not cosmos-sdk (I see it in node.go). Am I missing something?

Reply:
> having this error in the method is cleaner than adding logs for every return. There are also a few returns that don't error.
So you mean we return an error from this method and also from the others and refactor the comsos-sdk to act when an error happens?

> It also looks like this is called in cometbft not cosmos-sdk (I see it in node.go). Am I missing something?
this is called in cometBFT when you run plain comet (for example in e2e). But when you run an application, the cosmos-sdk overrides that server and calls this in there.

Reply:
It looks like so long as the RPC addr and GRPC addr in the config.toml is populated it will run it in CometBFT as well.

Ideally all go routines shut down gracefully when the node is shut down or errors. I would want the gRPC which is now something that the DA network will depend on to fail noisily. This could be with the consensus node also spinning down or something else that notifies the operator that there is a problem

Reply:
stopping the node is on the table? I guess it's good to stop the node if bridge nodes are connected to full nodes. But if they're connected to validators, it shouldn't kill the node as it will stop a validator